crates/llm-chain-sagemaker-endpoint/src/executor.rs

```diff
@@ -74,43 +77,120 @@ impl llm_chain::traits::Executor for Executor {
 
     fn tokens_used(
         &self,
-        _options: &Options,
-        _prompt: &Prompt,
+        options: &Options,
+        prompt: &Prompt,
     ) -> Result<TokenCount, PromptTokensError> {
-        // Not all models expose this information.
-        unimplemented!();
+        let tokenizer = self.get_tokenizer(options)?;
+        let input = prompt.to_text();
+
+        let tokens_used = tokenizer
+            .tokenize_str(&input)
+            .map_err(|_e| PromptTokensError::UnableToCompute)?
+            .len() as i32;
+        let max_tokens = self.max_tokens_allowed(options);
+        Ok(TokenCount::new(max_tokens, tokens_used)) 
     }
 
-    fn max_tokens_allowed(&self, _: &Options) -> i32 {
-        // Not all models expose this information.
-        unimplemented!();
+    fn max_tokens_allowed(&self, options: &Options) -> i32 {
+        let opts = self.cascade(Some(options));
+        let model = self.get_model_from_invocation_options(&opts);
+        model.context_window_size().unwrap_or_else(|| unimplemented!("This model does not expose max token allowed information."))
```

Comment:
If you are referring to the `i32` returned by the `max_tokens_allowed()`, I can't wrap it in a `Result` because I need to maintain the consistency of the `Executor` trait. It will break other Executors if I wrap it in a `Result`. 

If you are referring to the `Options<i32>` returned by the `context_window_size()` then yes I can change it.

Reply:
Got it, that's fine.